subiquity/common/filesystem/gaps.py

```diff
@@ -165,3 +166,24 @@ def largest_gap_size(device):
     if largest is not None:
         return largest.size
     return 0
+
+
+def movable_trailing_partitions_and_gap_size(partition):
+    pgs = parts_and_gaps(partition.device)
+    part_idx = pgs.index(partition)
+    trailing_partitions = []
+    in_extended = partition.flag == "logical"
+    for idx in range(part_idx + 1, len(pgs)):
+        pg = pgs[idx]
```

Comment:
Would something like the below make sense? It seems to me that you don't really need to store the index.
```python
for pg in pgs[part_idx + 1:]:
```
or
```python
from itertools import islice
for pg in islice(pgs, part_idx + 1, None):
```

Reply:
Oh yeah, totally.